app/jsx/components/Nav1Comp.jsx

```diff
@@ -1,20 +1,59 @@
 // ##### Navigation - Home Component ##### //
 
 import React from 'react'
+import { Link } from 'react-router'
 import Breakpoints from '../../js/breakpoints.json'
 
+// When running in the browser (and only then), include polyfill(s)
+if (!(typeof document === "undefined")) {
+  require('details-polyfill')
+}
+
 class Nav1Comp extends React.Component {
   state = { isOpen: true } // must init to something in case of server-side rendering
+
   componentWillMount() {
     if (typeof matchMedia != "undefined") {
       this.mq = matchMedia("(min-width:"+Breakpoints.screen2+")")
       this.mq.addListener(this.widthChange)
       this.widthChange()
     }
+    if (this.props.campuses) {
+      this.campusSelector = this.props.campuses.map(function(c, i) {
+          return c['id'] != "" && <Link key={i} to={"/unit/" + c['id']}>{c['name']}</Link>
+        })
```

Comment:
But what happens when the props change? componentWillMount() won't get called, and campusSelector could be out of date. I realize that's far-fetched, but my point is that code that generates UI elements, like this code, should be called as a result of render(). Since you've already got a function (campusListGenerator) that's called by render, I'd suggest incorporating this campusSelector generation directly in campusListGenerator.